request-manager/spec/node-check.go

```diff
@@ -0,0 +1,498 @@
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type SetsNotRenamedNodeCheck struct{}
+
+/* A single `sets` arg cannot be renamed. */
+func (check SetsNotRenamedNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	aliases := map[string]string{}
+	values := map[string]bool{}
+	for _, nodeSet := range node.Sets {
+		if nodeSet.Arg == nil || nodeSet.As == nil {
+			continue
+		}
+		if alias, ok := aliases[*nodeSet.Arg]; ok && alias != *nodeSet.As {
+			values[*nodeSet.Arg] = true
+		}
+		aliases[*nodeSet.Arg] = *nodeSet.As
+	}
+
+	if len(values) > 0 {
+		return DuplicateValueError{sequenceName, &node.Name, "sets -> arg", stringSetToArray(values), "value(s) renamed multiple times using `as`"}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type EachIfParallelNodeCheck struct{}
+
+/* If `parallel` is set, `each` must be set. */
+func (check EachIfParallelNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	/* `parallel` > 0. */
+	if node.Parallel != nil {
+		if node.Each == nil {
+			return MissingValueError{sequenceName, &node.Name, "each", "required when `parallel` field set"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type ValidParallelNodeCheck struct{}
+
+/* `parallel` > 0. */
+func (check ValidParallelNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.Parallel != nil {
+		if *node.Parallel == 0 {
+			return InvalidValueError{sequenceName, &node.Name, "parallel", []string{"0"}, "> 0"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type ConditionalNoTypeNodeCheck struct{}
+
+/* Conditional nodes may not specify a type. */
+func (check ConditionalNoTypeNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.IsConditional() {
+		if node.NodeType != nil {
+			return InvalidValueError{sequenceName, &node.Name, "type", []string{*node.NodeType}, "no value; conditional nodes may not specify a type"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type ConditionalHasIfNodeCheck struct{}
+
+/* `Conditional nodes must specify `if`. */
+func (check ConditionalHasIfNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.IsConditional() {
+		if node.If == nil {
+			return MissingValueError{sequenceName, &node.Name, "if", "required for conditional nodes"}
+		}
+	}
+
+	return nil
+}
+
+/* ========================================================================== */
+type ConditionalHasEqNodeCheck struct{}
+
+/* Conditional nodes must specify `eq`. */
+func (check ConditionalHasEqNodeCheck) CheckNode(sequenceName string, node NodeSpec) error {
+	if node.IsConditional() {
+		if len(node.Eq) == 0 {
```

Comment:
Similar to my comment about the `nodes` field in a sequence, if the node _has_ the `eq` field, but it's empty, does that cause a parsing error or result in an empty list? If the second, this field isn't really missing. eg:
```
if: foo
eq:
  # no values here
```